### ISSUE TITLE
Clarify the string format of AnySocketAddress

### ISSUE THREAD

**Issue by dom96** [wintercg/proposal-sockets-api#13]
Only `hostname:port` should be supported.

**Comment by mmastrac** [comment]
One consideration -- there may be some precedent for using `tcp://` prefixes in connection strings which may allow for additional embedder flexibility:

https://activemq.apache.org/uri-protocols

**Comment by jasnell** [comment]
If we make it so that the address supports absolute URL strings *or* `hostname:port` (https://github.com/wintercg/proposal-sockets-api/issues/16#issuecomment-1739809475) then I think we cover this. Host implementations would be free to decide which URL schemes they support in this. At a minimum, `tcp://` could be required along with the bare `hostname:port`.

A host could disambiguate the URL case by effectively allow-listing the set of URL protocol prefixes they support, e.g. if the runtime only supports `tcp://` and `http://`, then `12345:80` wouldn't ever be ambiguous. If the `12345` cannot be interpreted as one of the allowed protocols or as an IP address/hostname, then it is rejected.

**Comment by dom96** [comment]
Yeah, I'd say the port should be required for the protocol-less variant and that should resolve the ambiguity.

Shall we say the implementation needs to support:

* `tcp://` (where the port is required to be specified)
* `http://` and `https://` (where the port can be omitted and is 80, 443 respectively)
* protocol-less where both a hostname and port must be specified delimited by a `:`

If we agree I'll add this to the spec.

**Comment by jasnell** [comment]
I would say that we should support, at a minimum:

* `tcp://` where host and port are required
* All "special schemes" defined by the URL spec for which a default port is defined (https://url.spec.whatwg.org/#special-scheme). These include: `ftp` (port 21), `http` (port 80), `https` (port 443), `ws` (port 80), and `wss` (port 443).
* protocol-less, which is assumed to be equivalent to `tcp://`
